awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -410,6 +410,27 @@ def modify_associations(self, association_endpoint, new_association_list):
             else:
                 self.fail_json(msg="Failed to associate item {0}".format(response['json'].get('detail', response['json'])))
 
+    def copy_item(self, existing_item, name, item_type='unknown'):
```

Comment:
See comments below about merging more code in here but some cleanup on the variable names might be helpful. i.e. existing_item => item_to_copy, name => new_item_name, etc.

Reply:
Cleaned